src/app/css-variables.ts

```diff
@@ -1,56 +1,59 @@
- */
-// TODO: swap these into hooks
-export default function updateCSSVariables() {
-  observeStore(settingsSelector, (currentState, nextState, state) => {
-    if (!currentState) {
-      return;
-    }
-
-    if (currentState.itemSize !== nextState.itemSize) {
-      setCSSVariable('--item-size', `${Math.max(48, nextState.itemSize)}px`);
-    }
-    if (currentState.charCol !== nextState.charCol && !isPhonePortraitSelector(state)) {
-      setCSSVariable('--tiles-per-char-column', nextState.charCol);
-    }
-    if (
-      currentState.charColMobile !== nextState.charColMobile &&
-      // this check is needed so on start up/load this doesn't override the value set above on "normal" mode.
-      isPhonePortraitSelector(state)
-    ) {
-      setCSSVariable('--tiles-per-char-column', nextState.charColMobile);
-    }
+export function createItemSizeObserver(): StoreObserver<number> {
+  return {
+    id: 'item-size-observer',
+    getObserved: (rs) => settingsSelector(rs).itemSize,
+    sideEffect: ({ current }) => {
+      setCSSVariable('--item-size', `${Math.max(48, current)}px`);
+    },
+  };
+}
 
-    // Set a class on the body to control the theme. This must be applied on the body for syncThemeColor to work.
-    if ($featureFlags.themePicker && currentState.theme !== nextState.theme) {
-      const themeClass = `theme-${nextState.theme}`;
+export function createThemeObserver(): StoreObserver<{ theme: string; isPhonePortrait: boolean }> {
+  return {
+    id: 'theme-observer',
+    equals: deepEqual,
+    getObserved: (rs) => ({
+      theme: settingsSelector(rs).theme,
+      isPhonePortrait: isPhonePortraitSelector(rs),
+    }),
+    sideEffect: ({ current }) => {
+      // Set a class on the body to control the theme. This must be applied on the body for syncThemeColor to work.
+      const themeClass = `theme-${current.theme}`;
       document.body.className = themeClass;
-      syncThemeColor(isPhonePortraitSelector(state));
-    }
-  });
+      syncThemeColor(current.isPhonePortrait);
+    },
+  };
+}
 
-  // a subscribe on isPhonePortrait is needed when the user on mobile changes from portrait to landscape
-  // or a user on desktop shrinks the browser window below isphoneportrait threshold value
-  observeStore(isPhonePortraitSelector, (_prev, isPhonePortrait, state) => {
-    const settings = settingsSelector(state);
-    setCSSVariable(
-      '--tiles-per-char-column',
-      isPhonePortrait ? settings.charColMobile : settings.charCol,
-    );
-    syncThemeColor(isPhonePortrait);
-  });
+export function createTilesPerCharColumnObserver(): StoreObserver<number> {
+  return {
+    id: 'tiles-per-char-column-observer',
+    getObserved: (rs) =>
+      isPhonePortraitSelector(rs)
+        ? settingsSelector(rs).charColMobile
+        : settingsSelector(rs).charCol,
+    sideEffect: ({ current }) => {
+      setCSSVariable('--tiles-per-char-column', current);
+    },
+  };
+}
```

Comment:
I really like that we have drilled this down to be a single primitive value being observed. While it is likely not achievable everywhere definitely something to be on the look out for.

Reply:
Yeah this is great!